src/lustre/lustreDesugarIfBlocks.ml

```diff
@@ -0,0 +1,330 @@
+    )
+
+
+(** Converts an if block to a map of trees (creates a tree for each equation LHS) *)
+let if_block_to_trees ib =
+  let rec helper ib trees conds = (
+    match ib with 
+      | A.IfBlock (pos, cond, ni::nis, nis') -> (
+        match ni with
+          | A.Body (Equation (_, lhs, expr)) -> 
+          (* Update corresponding tree (or add new tree if it doesn't exist) *)
+          let trees = LhsMap.update lhs 
+            (fun tree -> match tree with
+              | Some tree -> Some (add_eq_to_tree (conds @ [(true, cond)]) expr tree)
+              | None -> Some (add_eq_to_tree (conds @ [(true, cond)]) expr (Leaf None)))
+            trees 
+          in
+          (helper (A.IfBlock (pos, cond, nis, nis')) trees conds)
+          (* Recurse, keeping track of our location within the if blocks using the 
+             'conds' list. *)
+          | A.IfBlock _ -> 
+            let* res = (helper ni trees (conds @ [(true, cond)])) in
+            (helper (A.IfBlock (pos, cond, nis, nis'))
+                   res
+                   conds)
+          (* Misplaced frame block, annot main, or annot property *)
+          | A.Body (Assert (pos, _)) 
+          | A.FrameBlock (pos, _, _, _)
+          | A.AnnotProperty (pos, _, _) 
+          | A.AnnotMain (pos, _) -> mk_error pos (MisplacedNodeItemError ni)
+        )
+      | A.IfBlock (pos, cond, [], ni::nis) -> (
+        match ni with
+          | A.Body (Equation (_, lhs, expr)) -> 
+            (* Update corresponding tree (or add new tree if it doesn't exist) *)
+            let trees = LhsMap.update lhs 
+              (fun tree -> match tree with
+                | Some tree -> Some (add_eq_to_tree (conds @ [(false, cond)]) expr tree)
+                | None -> Some (add_eq_to_tree (conds @ [(false, cond)]) expr (Leaf None))) 
+              trees 
+            in
+            (helper (A.IfBlock (pos, cond, [], nis)) trees conds)
+          (* Recurse, keeping track of our location within the if blocks using the 
+             'conds' list. *)
+          | A.IfBlock _ -> 
+            let* res = (helper ni trees (conds @ [(false, cond)])) in
+            (helper (A.IfBlock (pos, cond, [], nis))
+                   res
+                   conds)
+          (* Misplaced frame block, annot main, or annot property *)
+          | A.FrameBlock (pos, _, _, _)
+          | A.Body (Assert (pos, _)) 
+          | A.AnnotProperty (pos, _, _)
+          | A.AnnotMain (pos, _) -> mk_error pos (MisplacedNodeItemError ni)
+        )
+      (* We've processed everything in the if block. *)
+      | A. IfBlock (_, _, [], []) -> R.ok (trees)
+      (* shouldn't be possible *)
+      | _ -> assert false
+  ) in
+  (helper ib LhsMap.empty [])
+  
+(** Converts a tree of conditions/expressions to an ITE expression. *)
+let rec tree_to_ite pos node =
+  match node with
+    | Leaf Some expr -> expr
+    | Leaf None -> A.Ident(pos, HString.mk_hstring "ib_oracle")
```

Comment:
Define `HString.mk_hstring "ib_oracle"` as global constant in the module and use the constant everywhere. `fill_ite_with_oracles` depends on this constant.